### ISSUE TITLE
Meaning of vAc

### ISSUE THREAD

**Issue by jhlq** [iwhoppock/boris-algorithm#2]
Is vAc volts AC power? What is its physical interpretation in the algorithm?

**Comment by iwhoppock** [comment]
No, it is the alfven velocity divided by the speed of light, and you can
see this in the C code in the repo.

I am not sure I understand what you mean by the interpretation in the
algorithm. Look at equation 3.1 here: https://arxiv.org/pdf/1811.08873
you can go to the original paper by Boris which you can find in the
citations.

**Comment by jhlq** [comment]
Ok, thanks.
This article: https://edutinker.com/glossary/alfven-speed/
Says:
"The Alfvén speed is given by the equation:
V_A = B / (4πρ)^(1/2)
where B is the magnetic field, and ρ is the mass density of the plasma."
Is that what I should set vAc to?
Why did you divide by the speed of light?

**Comment by iwhoppock** [comment]
You divide velocity by c to make the code dimensionless, which allows for
normalization. Dig into the C code: You’ll see how things are set up (eg
mass is just 1 — normalized and dimensionless). It’s much more efficient
for the computer to not carry absurdly small numbers that can cause issues
with the machine precision. You can make the alfven speed / speed of light
ratio what ever you want, but it’s probably good to keep it physical and
non-relativistic. I believe Boris is not the standard for relativistic
particle tracing. Again, if you look in the code, it’s there. You can also
look at the blogs and get a more basic version of this. There is one linked
in the readme that is great.

**Comment by iwhoppock** [comment]
Here is this snippet from the dipole code:

The units are normalised: speed is in terms of the Alfven speed,
frequencies the proton cyclotron frequency, lengths the proton inertial
length, etc. To change to an alpha, replace mass with 4 and charge with 2.
Change the velocities gradually. This code implements the Boris algorithm
for non-relativistic particle tracing. The quantity vAc is the ratio of the
alfven speed by the speed of light (c).

**Comment by jhlq** [comment]
Alright, thanks.

Another question, can the Boris algorithm handle spatially varying magnetic fields? It doesn't take into account the gradient of B so I guess not? I set up a magnetic mirror simulation and particles don't bounce no matter how large B gets.

**Comment by iwhoppock** [comment]
Of course it can: Just put what ever magnetic field you want in. The
magnetic dipole repo is this exact thing.

**Comment by jhlq** [comment]
I cloned your dipole.py and plugged in a magnetic mirror field: https://gitlab.com/marcus.appelros/fusion/-/blob/main/dipole.py

There is no bounce..:
![noMirror](https://github.com/iwhoppock/boris-algorithm/assets/6324698/70d60582-a268-41e5-bbf8-bbad92bba588)

**Comment by jhlq** [comment]
The magnetic field I used was unphysical, when I fixed it so divergence is zero the Boris algorithm does bounce, but not at the correct place, also the point it bounces at shifts..:
![Screenshot from 2024-07-02 09-16-33](https://github.com/iwhoppock/boris-algorithm/assets/6324698/2250852c-5818-46fc-9fb9-2296e604e7c1)
It should bounce back and fourth between x=1 and -1

**Comment by jhlq** [comment]
I used the mirror field from this repo: https://github.com/BoschSamuel/Simulation-of-a-Tokamak-Fusion-Reactor/blob/master/B1.m
I turned off the central coil and scaled so B=1 at the origin.
Now it bounces between fixed points but it still doesn't bounce when B=2 but rather when B=45...
The code is here: https://gitlab.com/marcus.appelros/fusion/-/blob/main/Bosch/t0.jl
